asyncpg/protocol/coreproto.pyx

```diff
@@ -187,19 +187,23 @@ cdef class CoreProtocol:
         elif mtype == b'T':
             # Row description
             self.result_row_desc = self.buffer.consume_message()
+            self._push_result()
 
         elif mtype == b'E':
             # ErrorResponse
             self._parse_msg_error_response(True)
-
-        elif mtype == b'Z':
-            # ReadyForQuery
-            self._parse_msg_ready_for_query()
-            self._push_result()
+            # we don't send a sync during the parse/describe sequence
+            # but send a FLUSH instead. If an error happens we need to
+            # send a SYNC explicitly in order to mark the end of the transaction.
+            # this effectively clears the error and we then wait until we get a
+            # ready for new query message
+            self._write(SYNC_MESSAGE)
+            self._set_state(PROTOCOL_CANCELLED)
```

Comment:
It seems that `PROTOCOL_ERROR_CONSUME` would be more appropriate here.

Reply:
Sure, I can change it to that if preferred.